sample-skill/lambda/lambda_function.py

```diff
@@ -180,7 +181,44 @@ def get_course_id(course_name: str, username: str, token: str) -> str | None:
         if course["id"] in enrollments:
             valid_courses[course["name"].lower()] = course["id"]
 
-    return valid_courses.get(course_name)
+    return get_fuzzy_match(valid_courses, voice_input)
+
+
+def get_fuzzy_match(valid_courses: dict, voice_input: str) -> str | None:
+    """
+    Returns the course ID that best matches the voice input.
+
+    This function uses the matcher function to compare the voice input with the course
+    names and IDs in the valid_courses dictionary. It returns the course ID that has the
+    highest similarity ratio, or None if no match is found.
+
+    Args:
+        valid_courses (dict): A dictionary of course names and IDs.
+        voice_input (str): A string of the user's voice input.
+
+    Returns:
+        str | None: The course ID that best matches the voice input, or None if no match is found.
+    """
+    MIN_RATIO_FOR_CONSIDERATION = 0.6
+    possible_match = []
+
+    for name, course_id in valid_courses.items():
+        # Analize course name
+        voice_input_match = voice_input.lower()
+        ratio_name = matcher(None, voice_input_match, name).ratio()
```

Comment:
`voice_input` is already in lowercase
```suggestion
        ratio_name = matcher(None, voice_input, name).ratio()
```

Reply:
I would still sanitize it since this is an external input.

Reply:
yes, it makes sense